control-plane/api/v1alpha1/servicerouter_types.go

```diff
@@ -151,6 +151,9 @@ type ServiceRouteDestination struct {
 	NumRetries uint32 `json:"numRetries,omitempty"`
 	// RetryOnConnectFailure allows for connection failure errors to trigger a retry.
 	RetryOnConnectFailure bool `json:"retryOnConnectFailure,omitempty"`
+	// RetryOn is a flat list of conditions for Consul to retry requests based on the response from an upstream service.
+	// Refer to the valid conditions here: https://developer.hashicorp.com/consul/docs/connect/config-entries/service-router#routes-destination-retryon
+	RetryOn []string `json:"retryOn,omitempty"`
```

Comment:
Do we want to validate on the values that are specified here to fail if they use an unsupported condition?

Reply:
I'm on the fence about it. I'm very pro-validation, but since Envoy ultimately sets the API, I think I'll leave it as is to leave it flexible for Envoy updates. Let me know if you have strong opinions the other way.

Reply:
SGTM!! sorry i just saw this!